# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1305470341 25200
#      Sun May 15 07:39:01 2011 -0700
# Node ID 64193afe93d87c9c9fe5288d9de11891e833b910
# Parent  542a95cc1bd1ac6dc1a889660cafd67c16817286
runtests.m, rundemos.m: Remove final filesep() character from directory name
before looking it up in search patch.

* runtests.m, rundemos.m: Remove final filesep() character from directory name.

diff --git a/scripts/testfun/rundemos.m b/scripts/testfun/rundemos.m
--- a/scripts/testfun/rundemos.m
+++ b/scripts/testfun/rundemos.m
@@ -30,16 +30,17 @@
 function rundemos (directory)
 
   if (nargin == 0)
     dirs = strsplit (path (), pathsep ());
   elseif (nargin == 1)
     if (is_absolute_filename (directory))
       dirs = {directory};
     else
+      directory = regexprep (directory, ['\',filesep(),'$'], "");
       fullname = find_dir_in_path (directory);
       if (! isempty (fullname))
         dirs = {fullname};
       else
         error ("rundemos: DIRECTORY argument must be a valid pathname");
       endif
     endif
   else
diff --git a/scripts/testfun/runtests.m b/scripts/testfun/runtests.m
--- a/scripts/testfun/runtests.m
+++ b/scripts/testfun/runtests.m
@@ -30,16 +30,17 @@
 function runtests (directory)
 
   if (nargin == 0)
     dirs = strsplit (path (), pathsep ());
   elseif (nargin == 1)
     if (is_absolute_filename (directory))
       dirs = {directory};
     else
+      directory = regexprep (directory, ['\',filesep(),'$'], "");
       fullname = find_dir_in_path (directory);
       if (! isempty (fullname))
         dirs = {fullname};
       else
         error ("runtests: DIRECTORY argument must be a valid pathname");
       endif
     endif
   else
